### PR TITLE
fix: resolve 4 critical Java E2E pipeline bugs

### PR DESCRIPTION
**Pull request by mashraf-222**
## Summary

Fixes 4 critical bugs in the Java E2E optimization pipeline discovered during bug hunting against the aerospike-client-java project.

- **ANSI escape codes causing Rich console hangs** — Maven color output was being interpreted as Rich markup, causing terminal freezes
- **SQLite test result files not found crash** — When all behavioral tests fail, instrumentation hooks never fire, so SQLite files don't exist; the code crashed trying to compare them
- **AI hallucinating classes/methods in test generation** — Java testgen context only contained target code + same-file helpers, lacking imported type definitions that Python already provides

## Problems fixed

1. **Rich console hang**: Maven ANSI escape codes (color output) were being parsed by Rich's highlighter, causing the console to hang or produce garbled output during long Maven operations.

2. **FileNotFoundError on correctness verification**: After running candidate behavioral tests, the code went directly to SQLite file comparison with no early exit if all tests failed. Python uses in-memory comparison (no file dependency), but Java/JS require SQLite files that only exist when test instrumentation hooks fire — which doesn't happen when tests error out in setUp.

3. **Hallucinated constructors/methods**: Java testgen context only passed target code + same-file helpers. Unlike Python (which uses `get_imported_class_definitions()` to extract full class bodies from project modules), Java had no mechanism to provide imported type information, forcing the AI to guess constructor signatures and method APIs.

## Root causes

1. Rich's default highlighter interprets ANSI sequences in console output; Maven outputs colored text by default in non-batch mode.
2. The non-Python correctness verification path assumed SQLite files would always exist after behavioral tests, but instrumentation hooks only fire when tests actually execute their test bodies.
3. `get_code_optimization_context_for_language()` only assembled target code and same-file helpers for Java — no equivalent to Python's `get_imported_class_definitions()` existed.

## Solutions implemented

1. **NullHighlighter** on all Rich Console and RichHandler instances + **`-B` (batch mode) flag** on all Maven subprocess commands to suppress ANSI output at the source.

2. **Early exit guard** before SQLite comparison: checks `get_test_pass_fail_report_by_type()` and returns `get_results_not_matched_error()` when total passed tests is 0. Mirrors Python's `compare_test_results()` which returns `(False, [])` for empty results.

3. **New `get_java_imported_type_skeletons()`** function that resolves project-internal imports via `JavaImportResolver`, extracts class declarations + constructors + fields + public method signatures using `_extract_type_skeleton()`, and appends them to testgen context. Added `imported_type_skeletons` field to `CodeContext` dataclass and threaded it through to `CodeStringsMarkdown` for testgen.

## Code changes

| File | Change |
|------|--------|
| `codeflash/cli_cmds/console.py` | NullHighlighter on Console + RichHandler |
| `codeflash/cli_cmds/logging_config.py` | NullHighlighter on both RichHandlers |
| `codeflash/languages/java/test_runner.py` | `-B` flag on 4 Maven commands |
| `codeflash/languages/java/build_tools.py` | `-B` flag on 4 Maven commands |
| `codeflash/optimization/function_optimizer.py` | Early exit when 0 behavioral tests pass (non-Python) |
| `codeflash/languages/java/context.py` | New `get_java_imported_type_skeletons()` + helpers |
| `codeflash/languages/base.py` | `imported_type_skeletons` field on CodeContext |
| `codeflash/context/code_context_extractor.py` | Append skeletons to testgen context |

## Testing

**E2E validation (both pass):**
```
cd code_to_optimize/java/
CODEFLASH_CFAPI_SERVER=local CODEFLASH_AIS_SERVER=local uv run codeflash --file src/main/java/com/example/Fibonacci.java --function fibonacci --no-pr --verbose
CODEFLASH_CFAPI_SERVER=local CODEFLASH_AIS_SERVER=local uv run codeflash --file src/main/java/com/example/Calculator.java --function calculateStats --no-pr --verbose
```

- Fibonacci: optimization found, tests pass, correctness verified, 973x speedup
- Calculator.calculateStats: optimization found, tests pass, correctness verified, 3.6x speedup (exercises imported type resolution via MathHelpers dependency)

**Unit tests:** 72/72 Java context tests pass; 545 total pass (33 pre-existing failures in unrelated areas)

## Impact

- Unblocks Java E2E optimization on real-world projects with Maven builds
- Prevents crashes when generated tests fail during setUp (common with infrastructure-dependent code)
- Reduces AI hallucination in test generation by providing real type signatures

## Related

Companion PR in codeflash-internal: prompt updates for Bug 2 (mock infrastructure deps) and Bug 3 (use provided type signatures)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: codeflash/languages/java/context.py [hunk 1]
```diff
@@ -844,3 +849,162 @@ def extract_class_context(file_path: Path, class_name: str, analyzer: JavaAnalyz
     except Exception as e:
         logger.exception("Failed to extract class context: %s", e)
         return ""
+
+
+# Maximum token budget for imported type skeletons to avoid bloating testgen context
+IMPORTED_SKELETON_TOKEN_BUDGET = 2000
+
+
+def get_java_imported_type_skeletons(
```
Comment (misrasaurabh1) [inline codeflash/languages/java/context.py:858]: @mashraf-222 add tests for this please

**Comment by mashraf-222** [comment]
Added 13 unit tests covering `get_java_imported_type_skeletons()`, `_extract_public_method_signatures()`, and `_format_skeleton_for_context()`. Tests use the existing java_maven fixture project (Calculator importing MathHelper/Formatter). All 85 context tests pass.

**Comment by mashraf-222** [comment]
Added 14 more tests addressing coverage gaps:

**Bug 4 — early exit guard** (6 tests in `test_candidate_early_exit.py`):
- All tests failed / empty results / non-loop1 only → guard triggers (0 passed)
- Some tests passed / single pass among failures → guard does not trigger
- Mixed test types all failing → 0 across all types

**Bug 3 — edge cases** (8 tests in `test_context.py`):
- Wildcard imports skipped (class_name=None)
- Import to nonexistent class → None skeleton
- Skeleton output well-formed (braces check)
- Protected/package-private methods excluded
- Overloaded and generic method signatures
- Round-trip: `_extract_type_skeleton` → `_format_skeleton_for_context` with real fixture

Total: 99 tests pass across both files.

File: codeflash/languages/java/context.py (hunk 1 above)
Comment (mashraf-222) [reply codeflash/languages/java/context.py:858]: Added — 9 tests for `get_java_imported_type_skeletons()` covering internal resolution, method signatures, external import filtering, deduplication, empty input, token budget, wildcard imports, nonexistent class, and well-formed output. Plus tests for the helpers (`_extract_public_method_signatures`, `_format_skeleton_for_context`) and round-trip validation.
